### PR TITLE
Add modal component to the Style Guide

### PR DESCRIPTION
**Pull request by coderitual**
Add modal component to the style guide

Screenshot 2020-11-04 at 14 55 01
Screenshot 2020-11-04 at 14 51 26
Screenshot 2020-11-04 at 14 53 53

### CODE REVIEW

File: src/components/modal/Modal.jsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
+// @flow strict
+
+import * as React from 'react';
+import TopLayer from '../toplayer/TopLayer';
+import Overlay from '../overlay/Overlay';
+
+const OVERLAY_CLASS_NAME = 'sg-overlay';
+
+type PropsType = {
```
Comment (bartoszlorek) [inline src/components/modal/Modal.jsx:9]: we should think about these props, I believe most of them aren't needed

Comment (maciekbb) [reply src/components/modal/Modal.jsx:9]: I guess we should think if our goal here is to be backward-compatible or to provide the best API

Comment (bartoszlorek) [reply src/components/modal/Modal.jsx:9]: exactly, I think modal is not widely used because of this complex API

Comment (coderitual) [reply src/components/modal/Modal.jsx:9]: I agree with you @bartoszlorek  @maciekbb . Moving this component to the SG made me think about how many props are just passed down to the `TopLayer` and many of them are not working in modal in expected way (speaking gently). My initial idea was to keep the same interface for compatibility but if you're not happy from current implementation of Modal this is great case to improve that.

**Comment by coderitual** [comment]
@dsmileybrainly I will close this PR as modal in current form is not something we would be happy to have in SG. The only benefit would be removing it from `ui` folder.

**Pull request closed without merging**